labs/CompositionCollectionView/src/Layout.cs

```diff
@@ -0,0 +1,549 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+#nullable enable
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Numerics;
+using System.Threading.Tasks;
+#if WINAPPSDK
+using Microsoft.UI.Composition;
+using Microsoft.UI.Xaml;
+using Microsoft.UI.Xaml.Controls;
+using Microsoft.UI.Xaml.Hosting;
+#else
+using Windows.UI.Composition;
+using Windows.UI.Xaml;
+using Windows.UI.Xaml.Controls;
+using Windows.UI.Xaml.Hosting;
+using static CommunityToolkit.Labs.WinUI.CompositionCollectionView.AnimationConstants;
+#endif
+
+namespace CommunityToolkit.Labs.WinUI.CompositionCollectionView;
+public interface ILayout
+{
+    Action<ILayout, ILayout>? LayoutReplaced { get; set; }
+
+    void Activate(Panel panel);
+}
+
+#if !WINAPPSDK
+public abstract partial class Layout<TId, TItem> : ILayout, IDisposable

```

Comment:
`Layout` conflicts with the base `ItemsRepeater` `Layout` class and is too generic here (which is why you probably ran into immediate issues in WindowsAppSDK land). This should be more like `CompositionCollectionLayout` or something. Maybe we need a more concise name for the layout itself?

Reply:
Sounds good, also I'm flexible about naming in general, if someone suggests something that rolls off the tongue better than CompositionCollectionView I'm happy to rename it